raiden/transfer/architecture.py

```diff
@@ -282,7 +284,10 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
 
-class TransitionResult:
+ST = TypeVar('ST', bound=State)
+
+
+class TransitionResult(Generic[ST]):  # pylint: disable=unsubscriptable-object
```

Comment:
Nice use of generic. Note that at the next release of pylint we can remove this `disable` pragma. This is a bug on the pylint side: https://github.com/PyCQA/pylint/issues/2416